src/test/java/com/carmen/app/control/CarServiceTest.java

```diff
@@ -52,16 +55,16 @@ public void setUp() throws Exception {
 
 	@Test
 	public void testCarCreated() {
-		doNothing().when(this.persistenceService).createNew(car);
+		when(this.persistenceService.createNew(car)).thenReturn(car);
 		assertEquals(car, carService.createCar(car));
 		verify(this.persistenceService).createNew(car);
 	}
 
 	@Test
 	public void testGetCars() {
-		when(this.persistenceService.getAll("Car.FindCars", Car.class)).thenReturn(new ArrayList<Car>());
+		when(this.persistenceService.getAll(Mockito.eq(Car.class),Mockito.anyMap(),Mockito.anyString())).thenReturn(query);
 		List<Car> expectedCars = new ArrayList<Car>();
-		List<Car> cars = this.carService.getCars();
+		List<Car> cars = this.carService.getCars(1 , 2, "Seat", "brand");
 		assertEquals(expectedCars, cars);
 
 	}
```

Comment:
In general great work! One tip for your code: try to make methods as shorter, method with more that 10 lines are more difficult to read, I like to recommend Clean Code by Robert C. Martin is an excellent read on the topic, specially chapter 3 on functions I find it very interesting and can give you an interesting point of view.

Keep up the good work!